pcap/bpf_test.go

```diff
@@ -0,0 +1,65 @@
+package pcap
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+var (
+	snaplen = 65535
+	packet  = [...]byte{
+		0xff, 0xff, 0xff, 0xff, 0xff, 0xff, // dst mac
+		0x0, 0x11, 0x22, 0x33, 0x44, 0x55, // src mac
+		0x08, 0x0, // ether type
+		0x45, 0x0, 0x0, 0x3c, 0xa6, 0xc3, 0x40, 0x0, 0x40, 0x06, 0x3d, 0xd8, // ip header
+		0xc0, 0xa8, 0x50, 0x2f, // src ip
+		0xc0, 0xa8, 0x50, 0x2c, // dst ip
+		0xaf, 0x14, // src port
+		0x0, 0x50, // dst port
+	}
+	goodBPFFilter = "ip and tcp and port 80"
```

Comment:
possibly rename "matching" and "nonmatching", since the "bad" filter itself is actually a perfectly good filter that just doesn't match the above packet :)